src/mscorlib/ref/mscorlib.cs

```diff
@@ -7344,7 +7344,7 @@ public enum AssemblyNameFlags
     [System.Runtime.InteropServices.ComVisibleAttribute(true)]
     public class AssemblyNameProxy : System.MarshalByRefObject
     {
-        internal AssemblyNameProxy() { }
+        public AssemblyNameProxy() { }
```

Comment:
Right. we need `AssemblyName.GetAssemblyName` in System.Private.CoreLib. But the whole AssemblyNameProxy can live somewhere in corefx where it can be shared with .NET Native/CoreRT. It does not need to be here.